### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by algonell**
## Code changes

Fix typos in documentation, comments, etc.

### CODE REVIEW

File: docs/source/developer/patterns.rst [hunk 1]
```diff
@@ -121,7 +121,7 @@ Data Structures
 ---------------
 
 For public API, we have three options: JavaScript ``Array``,
-``IIterator``, and ``ReadonlyArray`` (an interface defined by
+``Iterator``, and ``ReadonlyArray`` (an interface defined by
```
Comment (JasonWeill) [inline docs/source/developer/patterns.rst:124]: `IIterator` is not a typo; the initial `I` indicates that this is an interface.

File: docs/source/developer/patterns.rst [hunk 2]
```diff
@@ -136,7 +136,7 @@ Prefer a ``ReadonlyArray``
 -  The values may need to be accessed randomly.
 -  A public attribute that is inherently static.
 
-Prefer an ``IIterator`` for:
+Prefer an ``Iterator`` for:
```
Comment (JasonWeill) [inline docs/source/developer/patterns.rst:139]: `IIterator` is not a typo; the initial `I` indicates that this is an interface.

File: jupyterlab/commands.py [hunk 3]
```diff
@@ -2095,7 +2095,7 @@ def _yarn_config(logger):
 
     Returns
     -------
-    {"yarn config": dict, "npm config": dict} if unsuccessfull the subdictionary are empty
+    {"yarn config": dict, "npm config": dict} if unsuccessful the subdictionary are empty
```
Comment (JasonWeill) [inline jupyterlab/commands.py:2098]:
```suggestion
    {"yarn config": dict, "npm config": dict}
    if unsuccessful, the subdictionaries are empty
```

**Review by JasonWeill — COMMENTED** [review]
Thank you so much for opening this issue! A couple of suggestions below. Also, I approved running the workflows. You may need to regenerate snapshots if any of the visual tests are broken by correcting typos.

File: docs/source/developer/patterns.rst (hunk 1 above)
Comment (algonell) [reply docs/source/developer/patterns.rst]: Had mixed feelings about those, thanks!

**Comment by algonell** [comment]
please update snapshots

**Comment by jtpio** [comment]
bot please update snapshots

**Review by krassowski — APPROVED** [review]
Thank you @algonell!

**Pull request merged**